internal/dsync/drwmutex.go

```diff
@@ -477,7 +476,12 @@ func lock(ctx context.Context, ds *Dsync, locks *[]string, id, source string, is
 		}
 	}
 
-	quorumLocked := checkQuorumLocked(locks, quorum) && locksFailed <= tolerance
+	//quorumLocked := checkQuorumLocked(locks, quorum) && locksFailed <= tolerance
+
+	//当set一共有4个节点，tolerance=1, quorum=3,当超时触发 lockFailed++是，lockFailed=1
+	//此时所有的节点响应都已经发送到 channel,前三个都是成功的，最后一个是失败的，lockFailed++, lockFailed=2 > tolerance，
+	//但是 lockSucceed >= quorum 是满足可以写入条件的，所以这里没有必要判断 lockFailed
+	quorumLocked := lockSucceed >= quorum
```

Comment:
do not need to change this logic here, comments should be in english.  You cannot just assume `lockSucceeded` to be the only value here.

We have to ensure tolerance for failed locks as well.